packages/vsce/__tests__/__e2e__/02_AddWiremockProfileAndListRegionsInCics.test.ts

```diff
@@ -0,0 +1,116 @@
+        const view = await zoweExplorer.openView();
+        cicsTree = await view.getContent().getSection("cics");
+    });
+
+    beforeEach(async () => {
+        // Expand the cics section
+        await cicsTree.click();
+        await cicsTree.expand();
+    });
+
+    describe("Adding Wiremock Profile In The Configuration File", () => {
+        it("Should Add Wiremock Profile", async () => {
+            // Add wiremock profile to the zowe.config.json
+            addNewProfile();
+            
+            // Find open editors
+            editorView = new EditorView();
+            const titles = await editorView.getOpenEditorTitles();
+
+            // Check zowe.config.json was opened - could check content here
+            expect(titles.some((title) => title.startsWith("zowe.config.json"))).is.true;
+
+            // Check if wiremock profile is added to the zowe.config.json
+            const editor = await editorView.openEditor("zowe.config.json") as TextEditor;
+            const isWmAvailable = await editor.getText()
+            expect(isWmAvailable.includes(text)).to.be.true;    
+        });
+    });
+
+    describe("Check For The Wiremock Profile And List The Regions", async () => {
+        let wiremockServer: TreeItem | undefined;
+
+        it("Quick Pick Option test", async () => {
+            // Click the plus icon in cics
+            const plusIcon = await cicsTree.getAction(`Create a CICS Profile`);
+            assert(typeof plusIcon !== "undefined");
+            await plusIcon.click();
+
+            // Find quickpick
+            quickPick = await InputBox.create();
+            const qpItems = await quickPick.getQuickPicks();
+            const qpLen = qpItems.length;
+        
+            // Selecting wiremock_server from the quickpick
+            let i, wiremockLabel;
+            for(i=0; i<qpLen; i++){
+                wiremockLabel = await qpItems[i].getLabel();
+                if(wiremockLabel.includes(text)){
+                    break;
+                }
+            }
+            expect(wiremockLabel).contains(text);
+            if(i<qpLen){
+              await quickPick.selectQuickPick(i);
+            }
+            cicsTree.takeScreenshot();
+          });
+
+        it("Wiremock Profile Available Under CICS Section Test", async ()=>{
+            // Checking if wiremock_server profile is available under the cics section
+            const potentialWireMockServer: TreeItem | undefined  = await cicsTree.findItem(text);
+            wiremockServer = potentialWireMockServer;
+            expect(potentialWireMockServer).exist;
+            cicsTree.takeScreenshot();
+          });
+
+        it("Should List The Regions Under Wiremock Profile", async () => {
+            // Expand the wiremock_server profile and list the regions under it
+            await wiremockServer?.click();
+            await wiremockServer?.collapse();
+            await sleep(3000);
```

Comment:
I was thinking about this over the weekend (sad, I know!)

It's fine to finish and merge this PR as it is, but I think on reflection we need to avoid sleeps. That's because these tests will run in line with the build - we'll only need to add 10x 3 second sleeps across our tests and the whole build will take 30 seconds longer than it potentially needs to.

So I think that in a follow-up PR it would be good to work out a way of instead of saying "wait for 3 seconds", we say "wait until x has happened", probably providing an anonymous function that checks that the wiremockServer node doesn't have any children or similar.

Do you think this is feasible - @almightyrush / @AndrewTwydell ? If we work this out nicely now we can reuse the same technique for future tests!

Reply:
Sounds good - a 100ms sleep is better for now - but as a strategy, avoiding sleeps will be great 👌